lib/glimr_api_client/pay_by_account.rb

```diff
@@ -0,0 +1,60 @@
+module GlimrApiClient
+  class PayByAccount < Base
+    class AccountNotFound < StandardError; end
+    class InvalidAccountAndConfirmation < StandardError; end
+    class InvalidAmount < StandardError; end
+    class GlobalStatusInactive < StandardError; end
+    class JurisdictionStatusInactive < StandardError; end
+    class UnspecifiedError < StandardError; end
+
+    attr_reader :request_body
+
+    def initialize(params)
+      @request_body = params
+    end
+
+    private
+
+    def check_request!
+      errors = []
+      [
+        :feeLiabilityId,
+        :pbaAccountNumber,
+        :pbaConfirmationCode,
+        :pbaTransactionReference
+      ].each do |required|
+        errors << required if request_body.fetch(required, nil).nil?
+      end
+
+      if request_body.fetch(:pbaTransactionReference, '').size > 240
+        errors << :pbaTransactionReferenceTooLong
+      end
+      raise RequestError, errors unless errors.empty?
+    end
+
+    def endpoint
+      '/pbapaymentrequest'
+    end
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    def re_raise_error(body)
+      error = body.fetch(:message, nil)
+      case body.fetch(:glimrerrorcode, nil)
```

Comment:
We're defaulting the fetched value to nil, but not handling nil in the case statement. Does that matter?